external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -2408,31 +2484,78 @@ fn lvalue(
+                sp(loc, E::LValue_::Var(None, name, None))
+            };
             let efields = match efields {
-                E::FieldBindings::Named(efields) => efields,
+                E::FieldBindings::Named(mut efields, ellipsis) => {
+                    if let Some(ellipsis_loc) = ellipsis {
+                        let mut fields = match field_info {
+                            FieldInfo::Empty => BTreeSet::new(),
+                            FieldInfo::Named(fields) => fields,
+                            FieldInfo::Positional(num_fields) => (0..num_fields)
+                                .map(|i| Field::add_loc(ellipsis_loc, format!("{i}").into()))
+                                .collect(),
+                        };
+
+                        for (k, _) in efields.key_cloned_iter() {
+                            fields.remove(&k);
+                        }
+
+                        let start_idx = efields.len();
+                        for (i, f) in fields.into_iter().enumerate() {
+                            efields
+                                .add(
+                                    Field(sp(ellipsis_loc, f.value())),
+                                    (start_idx + i, make_ignore(ellipsis_loc)),
+                                )
+                                .unwrap();
+                        }
+                    }
+
+                    efields
+                }
                 E::FieldBindings::Positional(lvals) => {
-                    let lvals = lvals.into_iter().enumerate().map(|(idx, l)| {
-                        let field_name = Field::add_loc(l.loc, format!("{idx}").into());
-                        (field_name, (idx, l))
-                    });
-                    UniqueMap::maybe_from_iter(lvals).unwrap()
+                    let fields = field_info.field_count();
+                    let missing = (fields as isize) - lvals.len() as isize;
+                    let mut expanded_lvals = UniqueMap::new();
+                    let mut idx = 0;
+                    for l in lvals.into_iter() {
+                        match l {
+                            E::Ellipsis::Binder(l) => {
+                                let field_name = Field::add_loc(l.loc, format!("{idx}").into());
+                                expanded_lvals.add(field_name, (idx, l)).unwrap();
+                                idx += 1;
+                            }
+                            E::Ellipsis::Ellipsis(eloc) => {
+                                for _ in 0..=missing {
+                                    let field = Field::add_loc(loc, format!("{idx}").into());
+                                    expanded_lvals.add(field, (idx, make_ignore(eloc))).unwrap();
+                                    idx += 1;
+                                }
+                            }
+                        }
+                    }
+                    expanded_lvals
```

Comment:
Same: is it possible to abstract/reuse this logic?

Reply:
Thanks for the suggestion. Was able to abstract and it's much cleaner now :)